lib/streamlit/runtime/connection_factory.py

```diff
@@ -41,64 +51,101 @@
 @gather_metrics("experimental_connection")
 @cache_resource
 def _create_connection(
-    connection_class: Type[ConnectionClass], name: str = "default", **kwargs
+    name: str, connection_class: Type[ConnectionClass], **kwargs
 ) -> ConnectionClass:
     """Create an instance of connection_class with the given name and kwargs.
 
     This function is useful because it allows us to @gather_metrics at a point where
     connection_class must be a concrete type. The public-facing connection API allows
     the user to specify the connection class to use as a string literal for convenience.
     """
-    return connection_class(connection_name=name, **kwargs)
 
+    if not issubclass(connection_class, BaseConnection):
+        raise StreamlitAPIException(
+            f"{connection_class} is not a subclass of BaseConnection!"
+        )
 
-# NOTE: Adding support for a new first party connection requires:
-#   1. Adding the new connection name and class to this function.
-#   2. Writing a new @overload signature mapping the connection's name to its class.
-def _get_first_party_connection(connection_name: str):
-    FIRST_PARTY_CONNECTIONS = {"snowpark", "sql"}
+    return connection_class(connection_name=name, **kwargs)
 
-    if connection_name == "snowpark":
-        return Snowpark
-    elif connection_name == "sql":
-        return SQL
+
+def _get_first_party_connection(connection_class: str):
+    if connection_class in FIRST_PARTY_CONNECTIONS:
+        return FIRST_PARTY_CONNECTIONS[connection_class]
 
     raise StreamlitAPIException(
-        f"Invalid connection {connection_name}. "
+        f"Invalid connection '{connection_class}'. "
         f"Supported connection classes: {FIRST_PARTY_CONNECTIONS}"
     )
 
 
 @overload
 def connection_factory(
-    connection_class: Literal["sql"],
-    name: str = "default",
-    autocommit: bool = False,
-    **kwargs,
+    name: str, connection_class: Literal["sql"], autocommit: bool = False, **kwargs
 ) -> SQL:
     pass
 
 
 @overload
 def connection_factory(
-    connection_class: Type[ConnectionClass], name: str = "default", **kwargs
+    name: str, connection_class: Literal["snowpark"], **kwargs
+) -> Snowpark:
+    pass
+
+
+@overload
+def connection_factory(
+    name: str, connection_class: Type[ConnectionClass], **kwargs
 ) -> ConnectionClass:
     pass
 
 
-def connection_factory(connection_class, name="default", **kwargs):
+@overload
+def connection_factory(
+    name: str, connection_class: Union[str, None], **kwargs
```

Comment:
super nit: we can use `Optional[str]` here

Reply:
Ah, yeah. There were previously three things in the union, so I preferred `Union[str, ConnectionClass, None]` to `Optional[Union[str, ConnectionClass]]`, but now that there are two it should definitely be an `Optional`